save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt

```diff
@@ -40,28 +40,73 @@ class OrganizationService(
     }
 
     /**
-     * Mark organization with [organizationName] as deleted
+     * Mark organization with [organization] as deleted
      *
-     * @param organizationName an [Organization]'s name to delete
+     * @param newStatus is new status for [organization]
+     * @param organization is organization in which the status will be changed
      * @return deleted organization
      */
     @Suppress("UnsafeCallOnNullableType")
-    fun deleteOrganization(organizationName: String): Organization = getByName(organizationName)
+    fun changeOrganizationStatus(organization: Organization, newStatus: OrganizationStatus): Organization = organization
         .apply {
-            status = OrganizationStatus.DELETED
+            status = newStatus
         }
         .let {
             organizationRepository.save(it)
         }
 
+    /**
+     * Mark organization [organization] as deleted
+     *
+     * @param organization an [Organization] to delete
+     * @return deleted organization
+     */
+    fun deleteOrganization(organization: Organization): Organization = if (!hasProjects(organization.name)) {
+        changeOrganizationStatus(organization, OrganizationStatus.DELETED)
+    } else {
+        organization
+    }
+
+    /**
+     * Mark organization with [organization] as created.
+     * If an organization was previously banned, then all its projects become deleted.
```

Comment:
is it requirement or some limitation?

Reply:
If the organization was ```banned``` by the administrator, then all its projects also immediately become ```banned```, and after it is ```restored```, we will not be able to restore all projects with the statuses that they had before. Therefore, such a solution is proposed so that the user can ```restore``` his own projects that he will need.

Reply:
of course, it is possible not to ban all its projects when the organization is banished, but to add more checks so that when the project is called, the organization is also astive, but are these additional checks needed at every step?

Reply:
got the problem, but maybe left them as banned instead of deleted?

Reply:
It's not always that all projects will be banned. 
At the same time, it makes no sense to unban an organization if all projects in it are banned. Here it is easier for the user to create a new one. Yes, and ```super_admin``` will know about it and if anything, it's easy to ban some project

Reply:
> It's not always that all projects will be banned.

```kotlin
    @Transactional
    fun banOrganization(organization: Organization): Organization {
        projectService.getAllByOrganizationName(organization.name).forEach {
            it.status = BANNED
            projectService.updateProject(it)
        }
        return changeOrganizationStatus(organization, OrganizationStatus.BANNED)
    }
```

When we ban organization, we ban all projects.
I guess we can't unban project if organization is banned (btw, maybe add such validation?), so project can be banned or deleted for banned organization. Maybe left status of project as is when organization is unbanned?

Reply:
discussed offline: owner of organization can restore deleted project, but can't unban project. And we don't store previous state before banning. So it's the easier solution: delete project after unbanning organization